_playwright-tests/tests/API/ListTemplatesWithFilters.spec.ts

```diff
@@ -0,0 +1,167 @@
+                    name: `${templateNamePrefix}-${randomName()}-1`,
+                    repositoryUuids: [`${repo1.uuid!}`],
+                    arch: 'x86_64',
+                    version: '8',
+                },
+            });
+            const fetchedTemplate1 = await new TemplatesApi(client).getTemplate({ uuid: template1.uuid! });
+            createdTemplates.push(fetchedTemplate1);
+
+            const template2 = await new TemplatesApi(client).createTemplate({
+                apiTemplateRequest: {
+                    name: `${templateNamePrefix}-${randomName()}-2`,
+                    repositoryUuids: [`${repo1.uuid!}`],
+                    arch: 'aarch64',
+                    version: '9',
+                },
+            });
+            const fetchedTemplate2 = await new TemplatesApi(client).getTemplate({ uuid: template2.uuid! });
+            createdTemplates.push(fetchedTemplate2);
+
+            const template3 = await new TemplatesApi(client).createTemplate({
+                apiTemplateRequest: {
+                    name: `${templateNamePrefix}-${randomName()}-3`,
+                    repositoryUuids: [`${repo2.uuid!}`],
+                    arch: 'x86_64',
+                    version: '10',
+                },
+            });
+            const fetchedTemplate3 = await new TemplatesApi(client).getTemplate({ uuid: template3.uuid! });
+            createdTemplates.push(fetchedTemplate3);
+
+            const template4 = await new TemplatesApi(client).createTemplate({
+                apiTemplateRequest: {
+                    name: `${templateNamePrefix}-${randomName()}-4`,
+                    repositoryUuids: [`${repo2.uuid!}`],
+                    arch: 'aarch64',
+                    version: '9',
+                },
+            });
+            const fetchedTemplate4 = await new TemplatesApi(client).getTemplate({ uuid: template4.uuid! });
+            createdTemplates.push(fetchedTemplate4);
+        });
+
+        await test.step('List templates by name', async () => {
+            const templates_name_search = await new TemplatesApi(client).listTemplates({
+                name: createdTemplates[0].name,
+            });
+            expect(templates_name_search.data).toHaveLength(1);
+        });
+
+        await test.step('List templates by architecture', async () => {
+            const templates_arch_search = await new TemplatesApi(client).listTemplates({
+                arch: 'x86_64',
+            });
+            expect(templates_arch_search.data).toHaveLength(2);
+        });
+
+        await test.step('List templates by version', async () => {
+            const templates_version_search = await new TemplatesApi(client).listTemplates({
+                version: '10',
+            });
+            expect(templates_version_search.data).toHaveLength(1);
+        });
+
+        await test.step('List templates by architecture, version, limit, offset, sortby', async () => {
```

Comment:
For this test.step, can you assert that the expected template is found, not just the count?